pkg/git/libgit2/managed/ssh.go

```diff
@@ -79,16 +82,17 @@ func registerManagedSSH() error {
 func sshSmartSubtransportFactory(remote *git2go.Remote, transport *git2go.Transport) (git2go.SmartSubtransport, error) {
 	return &sshSmartSubtransport{
 		transport: transport,
+		logger:    logr.Discard(),
 	}, nil
 }
 
 type sshSmartSubtransport struct {
 	transport *git2go.Transport
+	logger    logr.Logger
```

Comment:
Since we extract the logger from the context, I'd have said that we don't need to store the logger as a separate field. But since we can't change the signatures of the smart transport method to pass context, storing the logger in a field seems like the only way.
Let's have a comment here about why we are doing this instead of always extracting from context. Just as a design doc for future reference when we need to refactor this more.